scale/job/configuration/configuration/job_configuration.py

```diff
@@ -0,0 +1,394 @@
+        :type mode: string
+        """
+
+        if mode not in [MODE_RO, MODE_RW]:
+            raise Exception('%s is not a valid mode' % mode)
+
+        if name in self._job_task_workspace_names:
+            raise InvalidJobConfiguration('Duplicate workspace %s in job task' % name)
+        self._configuration['job_task']['workspaces'].append({'name': name, 'mode': mode})
+
+    def add_post_task_workspace(self, name, mode):
+        """Adds a needed workspace to this job's post task
+
+        :param name: The name of the workspace
+        :type name: string
+        :param mode: The mode of the workspace, either MODE_RO or MODE_RW
+        :type mode: string
+        """
+
+        if mode not in [MODE_RO, MODE_RW]:
+            raise Exception('%s is not a valid mode' % mode)
+
+        if name in self._post_task_workspace_names:
+            raise InvalidJobConfiguration('Duplicate workspace %s in post task' % name)
+        self._configuration['post_task']['workspaces'].append({'name': name, 'mode': mode})
+
+    def add_pre_task_workspace(self, name, mode):
+        """Adds a needed workspace to this job's pre task
+
+        :param name: The name of the workspace
+        :type name: string
+        :param mode: The mode of the workspace, either MODE_RO or MODE_RW
+        :type mode: string
+        """
+
+        if mode not in [MODE_RO, MODE_RW]:
+            raise Exception('%s is not a valid mode' % mode)
+
+        if name in self._pre_task_workspace_names:
+            raise InvalidJobConfiguration('Duplicate workspace %s in pre task' % name)
+        self._configuration['pre_task']['workspaces'].append({'name': name, 'mode': mode})
+
+    def configure_workspace_docker_params(self, workspaces):
+        """Configures the Docker parameters needed for each workspace in the job execution tasks
+
+        :param workspaces: A dict of all workspaces stored by name
+        :type workspaces: {string: :class:`storage.models.Workspace`}
+        """
+
+        for param in self._get_workspace_docker_params(self.get_pre_task_workspaces(), workspaces):
+            self.add_pre_task_docker_param(param)
+        for param in self._get_workspace_docker_params(self.get_job_task_workspaces(), workspaces):
+            self.add_job_task_docker_param(param)
+        for param in self._get_workspace_docker_params(self.get_post_task_workspaces(), workspaces):
+            self.add_post_task_docker_param(param)
+
+    def get_job_task_docker_params(self):
+        """Returns the Docker parameters needed for the job task
+
+        :returns: The job task Docker parameters
+        :rtype: [:class:`job.configuration.configuration.job_configuration.DockerParam`]
+        """
+
+        params = []
+        for param_dict in self._configuration['job_task']['docker_params']:
+            param = DockerParam(param_dict['flag'], param_dict['value'])
+            params.append(param)
+        return params
```

Comment:
Minor suggestion, but I usually write these types of blocks as follows when possible:
```python
params = self._configuration['job_task']['docker_params']
return [DockerParam(param_dict['flag'], param_dict['value']) for param_dict in params]
```